spec/java/util/stream/DoubleStream.main.lsl

```diff
@@ -0,0 +1,1266 @@
+        j += 1;
+    }
+
+
+    fun *.takeWhile (@target self: DoubleStream, predicate: DoublePredicate): DoubleStream
+    {
+        if (this.linkedOrConsumed)
+            _throwISE();
+
+        if (predicate == null)
+            _throwNPE();
+
+        if (this.length == 0)
+        {
+            val emptyStorage: array<double> = action ARRAY_NEW("double", 0);
+            result = new DoubleStreamAutomaton(state = Initialized,
+                storage = emptyStorage,
+                length = 0,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+        else
+        {
+            var takeLength: int = 0;
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 0, this.length, +1,
+                _takeWhile_loop(i, takeLength, predicate)
+            );
+
+            val newLength: int = takeLength;
+            val newStorage: array<double> = action ARRAY_NEW("double", newLength);
+
+            var j: int = 0;
+            action LOOP_FOR(
+                i, 0, takeLength, +1,
+                _copy_takeWhile_loop(i, j, newStorage)
+            );
+
+            result = new DoubleStreamAutomaton(state = Initialized,
+                storage = newStorage,
+                length = newLength,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+
+        this.linkedOrConsumed = true;
+    }
+
+
+    @Phantom proc _takeWhile_loop (i: int, takeLength: int, predicate: DoublePredicate): void
+    {
+        if (action CALL(predicate, [this.storage[i]]))
+            takeLength += 1;
+        else
+            action LOOP_BREAK();
+    }
+
+
+    @Phantom proc _copy_takeWhile_loop (i: int, j: int, newStorage: array<double>): void
+    {
+        newStorage[j] = this.storage[i];
+        j += 1;
+    }
+
+
+    fun *.asLongStream (@target self: DoubleStream): LongStream
+    {
+        if (this.length == 0)
+        {
+            val emptyArray: array<long> = action ARRAY_NEW("long", 0);
+            result = new LongStreamAutomaton(state = Initialized,
+                storage = emptyArray,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+        else
+        {
+            val newStorage: array<long> = action ARRAY_NEW("long", this.length);
+
+            var i: int = 0;
+            action LOOP_FOR(
+                i, 0, this.length, +1,
+                _convertDoubleStorageToLong_loop(i, newStorage)
+            );
+
+            result = new LongStreamAutomaton(state = Initialized,
+                storage = newStorage,
+                length = this.length,
+                closeHandlers = this.closeHandlers,
+            );
+        }
+    }
+
+
+    @Phantom proc _convertDoubleStorageToLong_loop(i: int, newStorage: array<long>): void
+    {
+        newStorage[i] = this.storage[i] as long;
+    }
+
+
+    fun *.asIntStream (@target self: DoubleStream): IntStream
```

Comment:
I cannot seem to find this method in `DoubleStream` nor in `DoublePipeline`. Where is it declared?
Also this does not work:
```java
DoubleStream.of(1, 2, 3).asIntStream(); // error: Cannot resolve method 'asIntStream' in 'DoubleStream'
```

Reply:
Fixed